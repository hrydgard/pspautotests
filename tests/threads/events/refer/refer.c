#include "../sub_shared.h"

SETUP_SCHED_TEST;

void testRefer(const char *title, SceUID flag, SceKernelEventFlagInfo *flaginfo) {
	int result = sceKernelReferEventFlagStatus(flag, flaginfo);
	if (result == 0) {
		schedf("%s: OK\n", title);
	} else {
		printf("%s: Failed (%X)\n", title, result);
	}
}

int main(int argc, char **argv) {
	SceUID flag = sceKernelCreateEventFlag("refer1", 0, 0, NULL);
	SceKernelEventFlagInfo flaginfo;

	// Crashes.
	//testRefer("NULL info", flag, NULL);
	testRefer("Normal", flag, &flaginfo);

	schedf("\nSizes:\n");
	int sizes[] = {0, 1, 2, 4, 8, 44, 48, 52, 56, 64, 128, 256, 1024, 0xFF00, 0xFF0000, 0x80000001, 0xCCCCCCCC, 0xFFFFFFFF};
	int i, result;
	for (i = 0; i < sizeof(sizes) / sizeof(sizes[0]); ++i) {
		flaginfo.size = sizes[i];
		result = sceKernelReferEventFlagStatus(flag, &flaginfo);
		schedf("  %08X => %08X (result=%08X)\n", sizes[i], flaginfo.size, result);
	}
	schedf("\n");
	flushschedf();

	sceKernelDeleteEventFlag(flag);

	testRefer("NULL", 0, &flaginfo);
	testRefer("Invalid", 0xDEADBEEF, &flaginfo);
	testRefer("Deleted", flag, &flaginfo);

	flushschedf();

	BASIC_SCHED_TEST("NULL",
		result = sceKernelReferEventFlagStatus(NULL, &flaginfo);
	);
	BASIC_SCHED_TEST("Refer other",
		result = sceKernelReferEventFlagStatus(flag2, &flaginfo);
	);
	BASIC_SCHED_TEST("Refer same",
		result = sceKernelReferEventFlagStatus(flag1, &flaginfo);
	);

	return 0;
}