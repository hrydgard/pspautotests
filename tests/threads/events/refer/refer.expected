Normal: OK

Sizes:
  00000000 => 00000000 (result=00000000)
  00000001 => 00000034 (result=00000000)
  00000002 => 00000034 (result=00000000)
  00000004 => 00000034 (result=00000000)
  00000008 => 00000034 (result=00000000)
  0000002C => 00000034 (result=00000000)
  00000030 => 00000034 (result=00000000)
  00000034 => 00000034 (result=00000000)
  00000038 => 00000034 (result=00000000)
  00000040 => 00000034 (result=00000000)
  00000080 => 00000034 (result=00000000)
  00000100 => 00000034 (result=00000000)
  00000400 => 00000034 (result=00000000)
  0000FF00 => 00000034 (result=00000000)
  00FF0000 => 00000034 (result=00000000)
  80000001 => 00000034 (result=00000000)
  CCCCCCCC => 00000034 (result=00000000)
  FFFFFFFF => 00000034 (result=00000000)

NULL: Failed (8002019A)
Invalid: Failed (8002019A)
Deleted: Failed (8002019A)
NULL: A1B1C2E1D1F2 (thread=800201B5, thread bits=00000000, main=8002019A)
Refer other: A1B1C2E1D1F2 (thread=800201B5, thread bits=00000000, main=00000000)
Refer same: A1B1C2E1D1F2 (thread=800201B5, thread bits=00000000, main=00000000)
